test/edu/csus/ecs/pc2/core/imports/LoadICPCTSVDataTest.java

```diff
@@ -111,5 +115,88 @@ private void checkIfGroupsAssigned(List<Account> accountList) {
             }
         }
     }
+    
+    private IInternalContest loadSampleContest(IInternalContest contest, String sampleName) throws Exception {
+        IContestLoader loader = new ContestSnakeYAMLLoader();
+        String configDir = getTestSampleContestConfigDirectory(sampleName);
+
+        try {
+            return loader.fromYaml(contest, configDir);
+        } catch (Exception e) {
+            e.printStackTrace(System.err);
+            throw e;
+        }
+    }
+
+    private String getTestSampleContestConfigDirectory(String contestName) {
+        return getTestSampleContestDirectory(contestName) + File.separator + IContestLoader.CONFIG_DIRNAME;
+    }
+
+    private String getTestSampleContestConfigFile(String contestName, String filename) {
+        return getTestSampleContestDirectory(contestName) + File.separator + IContestLoader.CONFIG_DIRNAME + File.separator + filename;
+    }
+
+    /**
+     * Test loading a teams.tsv twice.
+     * 
+     * The 2nd load of teams and groups, in particular groups, caused all accounts' groups to be "empty"
+     * on the Accounts tab.  See 
+     * 
+     * @throws Exception
+     */
+    public void testReLoadTSV() throws Exception {
+
+        String groupsFilename = getTestSampleContestConfigFile("mini", LoadICPCTSVData.GROUPS_FILENAME);
+        assertFileExists(groupsFilename);
+
+        IInternalContest contest = loadSampleContest(null, "mini");
+        assertNotNull(contest);
+        IInternalController controller = new SampleContest().createController(contest, true, false);
+        
+        LoadICPCTSVData loader = new LoadICPCTSVData();
+        loader.setContestAndController(contest, controller);
+        
+        boolean loaded = loader.loadFiles(groupsFilename, false, false);
+        assertTrue("Expecting min contest loaded", loaded);
 
+        assertEquals("Number of groups", 12, contest.getGroups().length);
+        assertEquals("Number of accounts", 167, contest.getAccounts().length);
+
+        loader.checkFiles(groupsFilename);
+
+        /**
+         * Second load of groups from groups.tsv
+         */
+        Group[] loadedGroups = ICPCTSVLoader.loadGroups(loader.getGroupsFilename());
+        Account[] accounts = ICPCTSVLoader.loadAccounts(loader.getTeamsFilename());
+
+        /**
+         * Groups from contest/model, the authoritative groups from model.
+         */
+        Group[] modelGroups = contest.getGroups();
+
+        assertEquals("Number of groups", 12, modelGroups.length);
+        assertEquals("Number of accounts", 151, accounts.length);
+
+        /**
+         * Second load of groups from groups.tsv
+         */
```

Comment:
This is a repeat of a comment above; that doesn't seem accurate.

Reply:
removed.